### PR TITLE
Upgrade Razor's Roslyn dependency to 2.0.0.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Added C# 7 test to validate questionable features work end-to-end.
- Had to add several explicit package references to let our VS specific packages work as expected.

#1046 

Working on making a similar change for MVC.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+@{
+    var nameLookup = new Dictionary<string, (string FirstName, string LastName, object Extra)>()
+    {
+        ["John Doe"] = ("John", "Doe", true)
+    };
+
+    int Sixteen = 0b0001_0000;
+    long BillionsAndBillions = 100_000_000_000;
+    double AvogadroConstant = 6.022_140_857_747_474e23;
+    decimal GoldenRatio = 1.618_033_988_749_894_848_204_586_834_365_638_117_720_309_179M;
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml]: Should we test one of these and a tuple in an implicit expression?

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+@{
+    var nameLookup = new Dictionary<string, (string FirstName, string LastName, object Extra)>()
+    {
+        ["John Doe"] = ("John", "Doe", true)
+    };
+
+    int Sixteen = 0b0001_0000;
+    long BillionsAndBillions = 100_000_000_000;
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml]: Could we use this in an expression (e.g. `@(100_000_000_000)`)

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml (hunk 1 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml]: Hmmmm, really good idea. Yes we should!

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    double AvogadroConstant = 6.022_140_857_747_474e23;
+    decimal GoldenRatio = 1.618_033_988_749_894_848_204_586_834_365_638_117_720_309_179M;
+}
+
+@if (nameLookup.TryGetValue("John Doe", out var entry))
```
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml]: Can we add some Html and some comments to the mix? Would be nice to see if that doesn't break things.

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml (hunk 2 above)
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml]: Looks like any complex double of any kind has never worked in implicit expression. Will add tests for doing this with explicit expressiong.

Bug for implicit expressions: https://github.com/aspnet/Razor/issues/1218

**Comment by NTaylorMullen** [comment]
🆙 📅

File: test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        if (entry.Extra is bool alive)
+        {
+            // Do Something
+        }
+    }
+    <p>
+        Here's a very unique number: @(1.618_033_988_749_894_848_204_586_834_365_638_117_720_309_179M)
+    </p>
+
+    <div>
```
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml]: Add an attribute with the expression?

**Pull request merged**